File: xcode-core/examples/typescript/simple_function.ts
/**
 * 简单函数 trace 示例
 * 用 console.log 模拟 trace
 */

function greet(name: string): string {
  return `Hello, ${name}!`;
}

function add(a: number, b: number): number {
  return a + b;
}

console.log(greet("World"));
console.log(add(1, 2));

File: xcode-core/examples/python/simple_function.py
"""简单函数 trace 示例：trace 一个 hello world 风格函数"""


def greet(name: str) -> str:
    """Simple greeting."""
    return f"Hello, {name}!"


def add(a: int, b: int) -> int:
    """Simple addition."""
    return a + b


if __name__ == '__main__':
    print(greet("World"))
    print(add(1, 2))
